barretenberg/cpp/src/barretenberg/translator_vm/translator_flavor.hpp

```diff
@@ -704,89 +680,17 @@ class TranslatorFlavor {
      */
     class ProvingKey : public ProvingKey_<FF, CommitmentKey> {
       public:
-        BF batching_challenge_v = { 0 };
-        BF evaluation_input_x = { 0 };
         ProverPolynomials polynomials; // storage for all polynomials evaluated by the prover
 
         // Expose constructors on the base class
         using Base = ProvingKey_<FF, CommitmentKey>;
         using Base::Base;
 
         ProvingKey() = default;
-        ProvingKey(const CircuitBuilder& builder)
-            : Base(compute_dyadic_circuit_size(builder), 0)
-            , batching_challenge_v(builder.batching_challenge_v)
-            , evaluation_input_x(builder.evaluation_input_x)
+        ProvingKey(const size_t dyadic_circuit_size, std::shared_ptr<CommitmentKey> commitment_key = nullptr)
+            : Base(dyadic_circuit_size, 0, std::move(commitment_key))
```

Comment:
Why move the commitment key?

Reply:
If a shared_ptr is passed as value to the constructor (as it is here) moving it is the right thing to do. Also, passing by value and moving is the right pattern for smart pointers (clangd-format suggests it).

Note that I'm not judging IF the constructor should take a commitment key or not! That is, whether the addition makes sense.